### ISSUE TITLE
Rules: use `–`[U+2013] (not `-`[U+002D]) to separate phrases

### ISSUE THREAD

**Issue by pygarap** [rvben/rumdl#408]
## Summary

Add a rule to enforce using `–`[U+2013] for phrase separation in prose, instead of `-`[U+002D] when written as `SPACE + dash + SPACE`.

Specifically, flag `SPACE + -`[U+002D]` + SPACE` in text content and auto-fix it to `SPACE + –`[U+2013]` + SPACE`.

## Scope

Apply to normal text content only.

Do not report or modify:
- Fenced and indented code blocks
- Inline code spans
- Link destinations (URLs), autolinks, and raw HTML blocks

## Auto-fix

When `--fix` (or `fmt`) is used, replace `SPACE + -`[U+002D]` + SPACE` with `SPACE + –`[U+2013]` + SPACE`.

Notes:
- `–`[U+2013] is multi-byte in UTF-8, so fix ranges must be computed correctly in byte offsets.

### Incorrect

```text
Alpha <SPACE>-`[U+002D]`<SPACE> Beta
```

### Correct

```text
Alpha <SPACE>–`[U+2013]`<SPACE> Beta
```

## Acceptance criteria

- `rumdl check` reports violations only in text content (not code, not link destinations).
- `rumdl check --fix` and `rumdl fmt` apply the replacement correctly and deterministically.
- Tests cover:
  - Basic detection
  - Skipping code spans and code blocks
  - Skipping link destinations
  - Correct UTF-8 byte range replacement

## References

- https://github.com/rvben/rumdl/blob/main/docs/RULES.md
- https://github.com/rvben/rumdl/blob/main/docs/global-settings.md
- https://github.com/rvben/rumdl/blob/main/docs/inline-configuration.md

**Comment by rvben** [comment]
Thanks for these suggestions, @pygarap. Prose style rules like en-dash usage, compound modifier hyphenation, and comma placement fall outside rumdl's scope. rumdl focuses on markdown structure and formatting, not English grammar or style.

For prose quality checks, dedicated tools like [Harper](https://github.com/Automattic/harper) or [Vale](https://github.com/errata-ai/vale) are a better fit. They work well alongside rumdl in CI or editor setups.

Closing this and the related issues #410, #411.
